Update best-practices.rst

Mrt134 opened the pull request:
Attempted to maintain the built look of this file.
Rewrote sections for clarity, fix mistakes, remove obsolete information.
Added more robust Sphinx/ReST markup.
Build test - no errors.

Make sure to follow and check these boxes before submitting a PR! Thank you.

- [x] Explain the feature
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

Mrt134 committed:
Update best-practices.rst

Attempted to maintain the built look of this file.
Rewrote sections for clarity, fix mistakes, remove obsolete information.
Added more robust Sphinx/ReST markup.
Build test - no errors.

skarekrow commented on doc/source/best-practices.rst:
```diff
@@ -1,56 +1,81 @@
+  jail. This helps avoid mistakes and easily identify jails.
 
-  This will help you avoid mistakes and easily identify jails.
+  Example: :samp:`iocage create -r 11.0-RELEASE -n ExampleJail`
```
We use `testjail` below, let's stay consistent 👍

skarekrow commented on doc/source/best-practices.rst:
```diff
@@ -1,56 +1,81 @@
+  more about VNET.
 
 **Don't overuse resource limiting!**
 
```
Another note that this is not present as of `0.9.9`

skarekrow commented on doc/source/best-practices.rst:
```diff
@@ -1,56 +1,81 @@
+**Use** :command:`iocage restart` **instead of start/stop**
 
-**Use the restart command instead of start/stop**
+  Always restart a jail with the :command:`iocage restart` command. This
```
`iocage restart -s`

Mrt134 committed:
Add review comments

I removed the reference to resource limiting entirely, as that seems far enough back in versions (and discouraged) that it's probably a better idea to not mention it at all.
